src/meshapi/views/query_api.py

```diff
@@ -18,128 +30,68 @@
 """
 
 
-def filter_model_on_query_params(
-    query_params: Dict[str, Any],
-    model: type[models.Model],
-    permitted_filters: Dict[str, Optional[str]],
-    subsitutions: Dict[str, str] = None,
-):
-    query_dict = {}
-    for k, v in query_params.items():
-        if v:
-            if subsitutions and k in subsitutions:
-                query_dict[subsitutions[k]] = v
-            else:
-                query_dict[k] = v
-
-    filter_args = {}
-    for k, v in query_dict.items():
-        if k in permitted_filters.keys():
-            if permitted_filters[k]:
-                filter_args[f"{k}__{permitted_filters[k]}"] = v
-            else:
-                filter_args[f"{k}"] = v
-
-    return model.objects.filter(**filter_args)
-
-
-BUILDING_FILTERS = {
-    "street_address": "icontains",
-    "zip_code": "iexact",
-    "city": "icontains",
-    "state": "icontains",
-    "bin": "iexact",
-}
-
-
-@extend_schema_view(
-    get=extend_schema(
-        tags=["Legacy Query Form"],
-        parameters=[query_form_password_param] + map_query_filters_to_param_annotations(BUILDING_FILTERS),
-        summary="Query & filter based on Building attributes. "
-        "Results are returned as flattened spreadsheet row style output",
-        auth=[],
-    ),
-)
-class QueryBuilding(generics.ListAPIView):
-    serializer_class = QueryFormSerializer
-    pagination_class = None
-    permission_classes = [permissions.LegacyMeshQueryPassword]
-
-    def get_queryset(self):
-        buildings = filter_model_on_query_params(
-            self.request.query_params,
-            Building,
-            BUILDING_FILTERS,
-        )
+class QueryFilterListAPIView(FilterRequiredListAPIView):
+    def get(self, request, *args, **kwargs):
+        maybe_auth = self.request.headers["Authorization"]
+        if maybe_auth != f"Bearer {os.environ.get('QUERY_PSK')}":
+            return Response({"detail": "Invalid Password"}, 403)
+        return super().get(request, *args, **kwargs)
```

Comment:
Why do we have to do this here?

Does `permission_classes = [permissions.LegacyMeshQueryPassword]` not work?